explorer/interpreter/value.h

```diff
@@ -341,31 +346,36 @@ class NominalClassValue : public Value {
   static constexpr llvm::StringLiteral BaseField{"base"};
 
   NominalClassValue(Nonnull<const Value*> type, Nonnull<const Value*> inits,
-                    std::optional<Nonnull<const NominalClassValue*>> base)
-      : Value(Kind::NominalClassValue),
-        type_(type),
-        inits_(inits),
-        base_(base) {}
+                    std::optional<Nonnull<const NominalClassValue*>> base,
+                    Nonnull<const NominalClassValue** const> class_value_ptr);
 
   static auto classof(const Value* value) -> bool {
     return value->kind() == Kind::NominalClassValue;
   }
 
   template <typename F>
   auto Decompose(F f) const {
-    return f(type_, inits_, base_);
+    return f(type_, inits_, base_, class_value_ptr_);
   }
 
   auto type() const -> const Value& { return *type_; }
   auto inits() const -> const Value& { return *inits_; }
   auto base() const -> std::optional<Nonnull<const NominalClassValue*>> {
     return base_;
   }
+  auto has_vtable() const -> bool { return !vtable().empty(); }
+  auto vtable() const -> const VTable&;
+  // Returns a pointer of pointer to the child-most class value. It is updated
+  // when `NominalClassValue`s are constructed to point to the right instance.
```

Comment:
I should note, though I'm suggesting changing the semantic, I'd expect this sentence to be on the constructor (so that callers of the constructor see it).

Reply:
Adjusted comments.